src/rdkafka.c

```diff
@@ -3491,7 +3492,15 @@ static void rd_kafka_query_wmark_offsets_resp_cb(rd_kafka_t *rk,
 
         offsets = rd_kafka_topic_partition_list_new(1);
         err = rd_kafka_handle_ListOffsets(rk, rkb, err, rkbuf, request, offsets,
-                                          NULL);
+                                          &actions);
+
+        if (actions & RD_KAFKA_ERR_ACTION_REFRESH) {
+                /* Remove its cache in case the topic isn't a known topic. */
+                rd_kafka_wrlock(rk);
+                rd_kafka_metadata_cache_delete_by_name(rk, state->topic);
```

Comment:
I think we can check here also that if cache is hinted for this topic, and in that case, do nothing as a refresh is already ongoing.

Reply:
Same for ListOffsets

Reply:
This is already ensured by `rd_kafka_metadata_cache_delete_by_name` because it does a rd_kafka_metadata_cache_find with `valid` equal to 1.